community/kernel/src/main/java/org/neo4j/kernel/NeoStoreDataSource.java

```diff
@@ -1382,6 +1388,14 @@ private StatementOperationParts buildStatementOperations(
         return parts;
     }
 
+    public void maintenance()
```

Comment:
I'm not sure putting `BufferingIdGeneratorFactory` in dependency resolver will work as it will overwrite the existing factory, which it cannot (think HA). But can try what happens.